lib/types/keypair.ex

```diff
@@ -32,6 +33,14 @@ defmodule Kadena.Types.KeyPair do
 
   def new(_args), do: {:error, [args: :not_a_list]}
 
+  @spec add_caps(caps :: clist(), keypair :: t()) :: validated_keypair()
+  def add_caps(caps, %__MODULE__{} = keypair) do
```

Comment:
@Odraxs @EdwinGuayacan IMO you should place `keypair` as the first argument of the function